### DISCUSSION TITLE
Questions about SideSetsFromNodeSetsGenerator

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#22553] [Q&A Meshing] [answered]
Dear all,
I am facing some issues with the SideSetsFromNodeSetsGenerator. We are trying to generate a sideset along a boundary of a model from some given input nodes. The model is 3D and contains also lower dimensional elements that are connected to the same boundary from which we are constructing the sides. The problem in itself is that the generated sideset also contains the elements from the lower dimensional material that we would like to exclude. Is there a way we can somehow blockwise the call of the meshgenerator to only specific blocks? Hope all makes sense, but in case I can replicate a simple model showing the issue we are facing.
Thanks for any feedback,
mauro

**Comment by GiudGiud** [comment]
Hello

No there is no block restriction (or even nodeset restriction) currently implemented in SideSetsFromNodeSetsGenerator. 

I'd recommend to use:
BreakBoundaryAlongBlocks
https://mooseframework.inl.gov/source/meshgenerators/BreakBoundaryOnSubdomainGenerator.html
or parsed expressions 
https://mooseframework.inl.gov/source/meshgenerators/ParsedGenerateSideset.html

to fix up the sideset after the conversion from the nodeset

Guillaume

**Comment by mcacace** [comment, marked as answer]
Thanks @GiudGiud. The BreakBoundaryAlongBlocks might be the solution. Quick question on top, is there a meshgenerators that could merge back the boundaries? Thanks!

**Reply by GiudGiud** [reply]
the RenameBoundaryGenerator may be used for merging
https://mooseframework.inl.gov/source/meshgenerators/RenameBoundaryGenerator.html

**Comment by mcacace** [comment]
Excellent! That made the trick. Thanks again @GiudGiud
